test: update slides (#9)
sfeir-open-source/sfeir-school-github-action-dev: docs/markdown/30-testing/01-others.md

# Testing Shell script with Bats

> ## Test folder

Reviewer: ```suggestion
Setup your repository by adding bats and utilities as submodules
```

Author: Done

> ```yaml

Reviewer: ```suggestion
```text
```

Author: Done

> .github/
>     actions/
>         runs-using-docker/
>             cleanup.sh
>             main.sh
>             ...

Reviewer: This refers to the implementation inside the school repository.
For the slide, can we use an fake action like

```suggestion
action.yaml
src/
    cleanup.sh
    main.sh
    setup.sh
```

Author: Done

test/
    bats/               <- submodule
    test_helper/
        bats-support/   <- submodule
        bats-assert/    <- submodule
    test.bats
    ...

> 

Reviewer: ```suggestion
```

Author: Done

```

> 

Reviewer: ```suggestion
```

Author: Done

##--##

# Testing Shell script with Bats

> ## **test.bats** 

Reviewer: ```suggestion
**test.bats** 
```

Slide Theme don't like `##` for some reasons 😄

Author: done

```

> setup() {

Reviewer: maybe add a line about how `setup` is run (or just a Speaker note to remember to tell it)

Author: Done + added some things about the teardown() method

    load 'test_helper/bats-support/load'
    load 'test_helper/bats-assert/load'

    DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )"
    export KEY=tata
    PATH="$DIR/../.github/actions/runs-using-docker:$PATH"
}

@test "check main.sh is working" {
    run main.sh toto
    assert_output 'Doing stuff... toto (tata)'
}

@test "check cleanup.sh is working" {
    run cleanup.sh toto
    assert_output 'Cleaning up... toto (tata)'
}

@test "check setup.sh is working" {
    run setup.sh toto
    assert_output 'Setting up... toto (tata)'
}
```

> - Easy way to writte action fgollowing the TDD method

Reviewer: ```suggestion
- Easy way to write action following the TDD method
```

Also, add pointer to TDD method, maybe?

Reviewer: This can be put before the code

Author: Done

- Required github submodule to work : 
    - `test/bats` 👉 [**Go to reference** 🔗](https://github.com/bats-core/bats-core.git)
    - `test/test_helper/bats-support` 👉 [**Go to reference** 🔗](https://github.com/bats-core/bats-support.git)

>     - `test/test_helper/bats-assert` 👉 [**Go to reference** 🔗](https://github.com/bats-core/bats-assert.git)

Reviewer: Move this part to the previous slide so that the flow will be 

- bats exist for shell testing
- setup bats into your action
- write a bats file
- run bats inside your GHA workflow

Author: Done
